api/v1/composition.go

```diff
@@ -121,17 +123,25 @@ type InputRevisions struct {
 	SynthesizerGeneration *int64 `json:"synthesizerGeneration,omitempty"`
 }
 
-func (i *InputRevisions) Equal(b InputRevisions) bool {
+func (i *InputRevisions) Less(b InputRevisions) bool {
 	if i.Key != b.Key {
-		return false
+		return false // shouldn't be possible
 	}
 	if (i.Revision == nil) != (b.Revision == nil) {
-		return false
+		return true
```

Comment:
Should this always be true? It feels off for this to pass when b == nil, but i is not

Reply:
My thinking was to fail open and just swap in the case that we can't determine ordering (one has revision one doesn't, etc.). But thinking more about it it's probably safer to fall back to the resource version. Thoughts?

Reply:
I'm okay with failing open here as long as we document the behavior. Though we don't have any logging here so potential issues could go unseen. Either way works for me.